### ISSUE TITLE
Cant take snapshot from ha

### ISSUE THREAD

**Issue by bedson87** [roleoroleo/yi-hack_ha_integration#129]
I've been having trouble with integration for some time (new home assistant version?). Lots of errors as below when trying to take a snapshot. Sometimes it works but usually writes an empty file. Everything else seems to work normally - recording to sd, preview, mqtt.

Model Suffix y211ga
Firmware Version 0.2.6
Yihack integr 0.4.5
Hass 2023.5.4

```
This error originated from a custom integration.

Logger: custom_components.yi_hack.camera
Source: custom_components/yi_hack/camera.py:249
Integration: Yi Cam with yi-hack (documentation, issues)
First occurred: 9:30:38 AM (2 occurrences)
Last logged: 9:29:16 PM

Fetch snapshot image failed from yi_hack_a2_c013c2_cam, falling back to FFmpeg; HTTPConnectionPool(host='192.168.10.28', port=8080): Read timed out.
```

**Comment by roleoroleo** [comment]
Does the snapshot work inside the web gui?

**Comment by bedson87** [comment]
Yes, both highres and lowres are displayed without any problems.

**Comment by roleoroleo** [comment]
HA shows a timeout problem.
Weird behavior...
Network problems?

**Comment by bedson87** [comment]
Nothing has changed with the network and there were no problems before.  Camera close to unifi ap.  Very good signal 🫢 After restarting the camera, the snapshot works but after some time it stops.  Could it be related to something else?  SD card?  Web ui takes a long time to load snapshot.
![Screenshot_20230604-224735](https://github.com/roleoroleo/yi-hack_ha_integration/assets/37672068/18c933f7-2783-4895-8970-3dde4d9b857f)

**Comment by roleoroleo** [comment]
Probably the problem is the cam.
Could you check the processes when the cam stops working properly?
Login to the cam with ssh and run `top` command.
And try to disable "Snapshot for recorded video".

**Comment by roleoroleo** [comment]
top is ok.
Now, try to run the snapshot manually:
`imggrabber -m y211ga -r high -w`

**Comment by bedson87** [comment]
umm
![Screenshot_20230606_212229](https://github.com/roleoroleo/yi-hack_ha_integration/assets/37672068/065cca5a-4523-4983-a324-805d41dde48d)

**Comment by roleoroleo** [comment]
It should be ok.
Try to redirect the command to a file and check if it's a jpeg image:
`imggrabber -m y211ga -r high -w > /tmp/sd/test.jpg`

**Comment by bedson87** [comment]
Thanks, it works. Strange thing - with HA once it works and 10 times it fails. I'll try to re-integrate or something.

**Comment by roleoroleo** [comment]
Check how long it takes. Maybe I could increase the timeout.

**Comment by bedson87** [comment]
I tried calling single snapshot from dev tools with HA and crash. Is there any good way to measure the time it takes to take a snapshot? Or is it possible to edit the config file for the test somewhere and change the integration timeout?

**Comment by roleoroleo** [comment]
HTTP_TIMEOUT is a const defined in const.py
Now it's set to 10 seconds.

Try to change it and restart ha to reload the integration.

**Comment by bedson87** [comment]
Now everything works. I didn't change the timeout. I'm following and thanks for your help and tips!
